pentasharp/Controllers/BusinessTaxiCompanyController.cs

```diff
@@ -0,0 +1,153 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Models.Enums;
+using WebApplication1.Filters;
+using pentasharp.Data;
+using pentasharp.Services;
+using pentasharp.Models.TaxiRequest;
+using pentasharp.Models.Utilities;
+
+namespace WebApplication1.Controllers
+{
+    [Route("Business/TaxiCompany")]
+    public class BusinessTaxiCompanyController : Controller
+    {
+        private readonly ITaxiService _taxiService;
+        private readonly IDriverService _driverService;
+        private readonly IHttpContextAccessor _httpContextAccessor;
+        private readonly AppDbContext _context;
+
+        public BusinessTaxiCompanyController(
+            ITaxiService taxiService,
+            IDriverService driverService,
+            IHttpContextAccessor httpContextAccessor,
+            AppDbContext context)
+        {
+            _taxiService = taxiService;
+            _driverService = driverService;
+            _httpContextAccessor = httpContextAccessor;
+            _context = context;
+        }
+
+        [HttpGet("GetCompany")]
+        public IActionResult GetCompany()
+        {
+            var userId = GetCompanyId();
+            if (!userId.HasValue)
```

Comment:
as per other comments, move logic to the service of this endpoint and all others in this controller